README.md

```diff
@@ -8,92 +8,420 @@
+
+class Profile extends React.Component {
+  render () {
+    const {userId} = this.props;
+    const profile = userStore.find((user) => user.id === userId);
+
+    if (!profile) {
+      return (
+        <div>Loading...</div>
+      );
+    }
+
+    return (
+      <dl>
+        <dt>First Name</dt>
+        <dd>{profile.firstName}</dd>
+        <dt>Last Name</dt>
+        <dd>{profile.lastName}</dd>
+        <dt>Bio</dt>
+        <dd>{profile.bio}</dd>
+      </dl>
+    );
+  }
+}
+```
+
+We're almost there now. Currently we assume that the user that should be
+rendered is available in the store at the point when we render the
+`<Profile />` component.
+
+This might not always be the case. E.g. our users are most likely going to be
+loaded from some sort of API. What if we render the component before 
+respective HTTP request receives a response? In this case, our profile
+component would simply display `Loading...` forever.
+
+We therefore need to notify subscribed components about eventual state changes.
+This sounds conceptually very similar to an event emitter and in fact that's
+exactly what we're going to implement!
+
+We can either inherit from some event emitter (e.g.
+`require('events').EventEmitter`) or we can implement our own. To recap the
+inner-workings of how an event emitter works, let's implement our own for now!
+
+## Turning our store into an `EventEmitter`
+
+Our store needs to expose an API for registering components that depend on its
+state. Analogous to an `EventEmitter`, we call the respective method
+`addListener` and `removeListener`.
+
+Both methods accept a listener function that will be called with the updated
+state object whenever the store's state changes.
+
+To start off, we need to store an array of those listener functions on the
+initiated store object:
 
-  removeListener(id) {
-    this.listeners = [
-      ...this.listeners.slice(0, id),
-      ...this.listeners.slice(id + 1)
-    ]
+```js
+class UserStore {
+  constructor (initialState) {
+    this.state = initialState;
+
+    // Our listener functions will be stored on UserStore#listeners.
+    this.listeners = [];
   }
 
-  updateState(state) {
-    this.state = state
+  // ...
+}
+```
+
+Now components can register themselves using `UserStore.addListener`. The
+`addListener` function is simply going to add the listener to the `listeners`
+array.
 
-    this.listeners.forEach(l => l())
+```js
+class UserStore {
+  // ...
+  addListener (listener) {
+    this.listeners.push(listener);
   }
 }
+```
 
-// stores/togglerStore.js
-const Store = require('./Store')
+The `UserStore` iterates through all registered event listeners whenever a
+state change occurs. All listeners will be called with the updated state as a
+first argument:
 
-const initialState = {
-  on: false
+```js
+class UserStore {
+  // ...
+  setState (state) {
+    this.state = state;
+    for (let listener of this.listeners) {
+      listener.call(null, state);
```

Comment:
Do we need `.call()`? I think we can just run the function as-is: `listener(state)`